sources/azuregit-source/src/azuregit.ts

```
@@ -0,0 +1,197 @@
+    const httpClient = axios.create({
+      baseURL: `https://dev.azure.com/${config.organization}/${config.project}/_apis`,
+      timeout: 10000, // default is `0` (no timeout)
+      maxContentLength: Infinity, //default is 2000 bytes
+      params: {
+        'api-version': version,
+      },
+      headers: {
+        Authorization: `Basic ${config.access_token}`,
+      },
+    });
+    const graphVersion = config.graph_version ?? DEFAULT_GRAPH_VERSION;
+    const graphClient = axios.create({
+      baseURL: `https://vssps.dev.azure.com/${config.organization}/_apis/graph`,
+      timeout: 10000, // default is `0` (no timeout)
+      maxContentLength: Infinity, //default is 2000 bytes
+      params: {
+        'api-version': graphVersion,
+      },
+      headers: {
+        Authorization: `Basic ${config.access_token}`,
+      },
+    });
+
+    AzureGit.azureGit = new AzureGit(httpClient, graphClient, logger);
+    return AzureGit.azureGit;
+  }
+
+  async checkConnection(): Promise<void> {
+    try {
+      const iter = this.getRepositories();
+      await iter.next();
+    } catch (err: any) {
+      let errorMessage = 'Please verify your access token is correct. Error: ';
+      if (err.error_code || err.error_info) {
+        errorMessage += `${err.error_code}: ${err.error_info}`;
+        throw new VError(errorMessage);
+      }
+      try {
+        errorMessage += err.message ?? err.statusText ?? wrapApiError(err);
+      } catch (wrapError: any) {
+        errorMessage += wrapError.message;
+      }
+      throw new VError(errorMessage);
+    }
+  }
+
+  async *getRepositories(): AsyncGenerator<Repository> {
+    const res = await this.httpClient.get<RepositoryResponse>(
+      'git/repositories'
+    );
+    for (const item of res.data.value) {
+      const branchResponse = await this.httpClient.get<BranchResponse>(
+        `git/repositories/${item.id}/stats/branches`
+      );
+      if (branchResponse.status === 200) {
```

In general, I wonder if we should move the axios calls into their own client class where we can handle non 200 responses and rate limiting backoff glabally for the azuregit source. It feels a little weird that we are just ignoring all our failure response. @cjwooo any ideas?

---

Yea we should not be ignoring error responses. We should at least be surfacing the error to the user.